CitationStyleLanguagePlugin.php

```diff
@@ -583,15 +583,23 @@ public function getCitation(PKPRequest $request, Submission $submission, string
                     $additionalMarkup = [
                         'DOI' => [
                             'function' => function ($item, $renderedValue) {
-                                return '<a href="https://doi.org/' . $item->DOI . '">' . $renderedValue . '</a>';
+                                $prefixPattern = '/.*DOI:/i';
+                                if (preg_match($prefixPattern, $renderedValue, $prefixMatches)) {
+                                    $renderedValue = preg_replace($prefixPattern, '', $renderedValue);
+                                }
+                                $suffixPattern = '#'.$item->DOI.'(.*)$#';
```

Comment:
If the `$item->DOI` has an unexpected/special character, an error might happen.

Reply:
Just to complement, when you're building dynamic regular expressions it's needed to use the `preg_quote()` (e.g. `preg_quote($item->DOI, '#')`).